src/Diagnostics.DataProviders/DataProviders/KustoDataProvider.cs

```diff
@@ -352,14 +380,31 @@ public async Task<string> GetAggHighPerfClusterNameByStampAsync(string stampName
 
         public async Task<DataTable> ExecuteQueryOnHighPerfClusterWithFallback(string aggQuery, string fallbackQuery, string stampName, string requestId = null, string operationName = null) 
         {
-            string cluster = await GetAggHighPerfClusterNameByStampAsync(stampName);
-            if (string.IsNullOrWhiteSpace(cluster))
+            operationName = operationName ?? GenerateOperationName();
+            string highPerfCluster = await GetAggHighPerfClusterNameByStampAsync(stampName);
+            string cluster = await GetClusterNameFromStamp(stampName);
+            await AddQueryInformationToMetadata(query: fallbackQuery, cluster: cluster, operationName: operationName + " (fallback)");
+
+            if (string.IsNullOrWhiteSpace(highPerfCluster))
             {
-                return await ExecuteQuery(fallbackQuery, stampName, requestId, operationName);
+                return await ExecuteQuery(fallbackQuery, stampName, requestId, operationName, addMetaData: false);
             }
             else
             {
-                return await ExecuteClusterQuery(aggQuery, cluster, _kustoMap.MapDatabase(_configuration.DBName) ?? _configuration.DBName, requestId, operationName);
+                return await ExecuteClusterQuery(aggQuery, highPerfCluster, _kustoMap.MapDatabase(_configuration.DBName) ?? _configuration.DBName, requestId, operationName, addMetaData: false);
+            }
+        }
+
+
+        private string GenerateOperationName()
+        {
+            if (_detectorId != null)
+            {
+                return _detectorId + $"_{unnamedOperationCount++}";
```

Comment:
Do you think we should have a delimiter other than "_" given that underscore is used commonly while naming something? 

It can help us in future analytics if we desire to.

Reply:
I see, I am thinking the pattern like "detectorName_N" as the query name for the unnamed queries, what about "detectorName.N"? I think we cannot use "detectorName;N" as the client activity id uses semicolon as delimiter already.

Reply:
How about a "|" symbol? Just want to avoid using something a real detector author may use and cause conflict with our reporting in future

Reply:
I see, if so what about "detectorName.N(auto generated)"?

Reply:
discussed offline and decided to apply the pattern "detectorId.unnamedQueryX"